core/src/main/scala/cromwell/util/JsonEditor.scala

```diff
@@ -88,11 +88,25 @@ object JsonEditor {
   def logs(json: Json): ErrorOr[Json] = {
     val inputsAndOutputs = NonEmptyList.of("outputs", "inputs")
     val shardAttemptAndLogsFields = NonEmptyList.of("shardIndex", "attempt", "stdout", "stderr", "backendLogs")
+
+    def removeSubworkflowCalls(calls: JsonObject): JsonObject = calls.filter {
+      // All calls within this array are assumed to have the same "shape": either they are subworkflows or regular jobs.
+      // So this only looks at the first element of the call array to determine whether this member of the containing
+      // object should be filtered as a subworkflow.
+      case (_, json) => ! json.asArray.get.head.asObject.exists(_.contains(subWorkflowIdKey))
+    }
+
     for {
-      wf <- updateWorkflowCallsJson(json, (_, _) => None) // Deletes all subworkflows found in the input by always returning `None`.
+      workflowWithSubworkflowsUnexpanded <- unexpandSubworkflows(json)
+      calls <- callsObject(workflowWithSubworkflowsUnexpanded)
+      callsWithSubworkflowsRemoved = calls map removeSubworkflowCalls
+      workflowWithSubworkflowsRemoved = callsWithSubworkflowsRemoved match {
```

Comment:
I'm not sure I'm getting this right:
1. So, at first we unexpand subworkflows by replacing `subWorkflowMetadata` objects with `subWorkflowId` in `calls`.
2. Then we remove all elements of `calls` array, which contain subWorkflowId.

In this case why cannot we skip unexpansion and just do `json.asArray.get.head.asObject.exists(_.contains(subWorkflowMetadataKey))` instead of `json.asArray.get.head.asObject.exists(_.contains(subWorkflowIdKey))`

Reply:
yes you're right I should do that here 👍

Reply:
though as a consequence I will need to add back some of the error checking that this code was previously relying on the subworkflow unexpansion to provide